tests/testthat/test-rapporttekst-funksjoner.R

```diff
@@ -113,3 +143,44 @@ test_that("akse_tall_format() gir ut en tom tekstvektor hvis den tar
   tall_format = akse_tall_format()
   expect_identical(tall_format(integer()), character())
 })
+
+# Testing av prosent()
+
+test_that("prosent() gir ut tankestrek når «x»-argumentet inneholder NA", {
+  expect_identical(prosent(NA_real_), "\\textendash{}")
+})
+
+test_that("prosent() gir ut riktig verdi når kun «x»-argumentet er gitt", {
+  expect_identical(prosent(numeric()), character(0))
+  expect_identical(prosent(0.1234), "{\\numprint{12}}\\prosent")
+  expect_identical(
+    prosent(c(0.1234, 0.31415)),
+    c("{\\numprint{12}}\\prosent", "{\\numprint{31}}\\prosent")
```

Comment:
Her (og i andre testar) har du hardkoda at `prosent()` skal gje ut `\numprint{}`-kommandoar med formaterte talverdiar. Det står det ingenting i dokumentasjonen om, og det er langt frå sikkert at han skal gjera dette i framtida. Det er altså ein unødvendig og uheldig avhengnad til `num()` sin gjeldande funksjonalitet i testane.

(Akkurat som funksjonar skal vera mest mogleg uavhengig av *gjeldande* åtferd til andre funksjonar, gjeld det same testar. Generelt er det slik at ein funksjon berre er ansvarleg for *si eiga åtferd*, og skal berre vera avhengig av at andre funksjonar gjer jobben sin (kva denne er, er definert i dokumentasjonen), ikkje *korleis* dei gjer jobben sin. Eks. er `num()` sin jobb å gje ut ein kvalreg-klasse-kompatibel LaTeX-kommando for formatering av tal med gitt # desimalar. Om dette skjer via `\numprint{}` eller noko anna, er berre noko `num()` skal veta inni seg, og ikkje dela med andre.)

Det `prosent()` *skal* gjera, er å gje ut ein vektor der elementa er lik resultatet av `num()`, men med `\prosent` hekta på. Det er *det* testane skal testa. Dette gjer òg at du får enklare (og gjerne færre) testar.

Reply:
Ja, forresten, ein må sjølvsagt òg testa at `NA`-verdiar vert handterte rett. Altså, viss ein har ein `NA`-verdi, som `num()` no gjev ut som `–`, skal `prosent()` òg gje han ut *likt som `num()` gjer* (og altså ikkje som `–\prosent`, slik ei naiv implementering ville gitt).

Reply:
Dette må òg fungera rett viss berre *nokre* av elementa i vektoren er `NA`.